flower/utils/results/result.py

```diff
@@ -14,12 +14,12 @@ def __init__(
         *,
         task_id: str,
         status: str,
-        date_done: str,
+        date_done: str | datetime.datetime,
```

Comment:
Convenience change to `__init__()` so that different backends can more easily initialize a `Result(...)` instance despite the backend subclasses representing timestamps using different data types.